src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/AccessorBuilder.cs

```diff
@@ -33,9 +34,43 @@ public static Getter CreateGetter(MemberInfo memberInfo)
         {
             if (memberInfo is PropertyInfo propInfo)
             {
-                var createGetterGeneric = s_createGetterInternal.MakeGenericMethod(propInfo.DeclaringType!, propInfo.PropertyType).CreateDelegate<Func<PropertyInfo, Getter>>();
-                Getter accessor = createGetterGeneric(propInfo);
-                return accessor;
+                Type declaringType = propInfo.DeclaringType!;
+                Type propertyType = propInfo.PropertyType!;
+
+                if (declaringType.IsGenericType && declaringType.GetGenericTypeDefinition() == typeof(KeyValue<,>))
+                {
+                    if (propInfo.Name == "Key")
+                    {
+                        return (obj) =>
+                        {
+                            return ((IKeyValue)obj).Key;
+                        };
+                    }
+                    else
+                    {
+                        return (obj) =>
+                        {
+                            return ((IKeyValue)obj).Value;
+                        };
+                    }
+                }
+
+                // If either of the arguments to MakeGenericMethod is a valuetype, this is going to cause JITting.
+                // Only JIT if dynamic code is supported.
+                if (RuntimeFeature.IsDynamicCodeSupported || (!declaringType.IsValueType && !propertyType.IsValueType))
+                {
+                    var createGetterGeneric = s_createGetterInternal.MakeGenericMethod(declaringType, propertyType).CreateDelegate<Func<PropertyInfo, Getter>>();
+                    Getter accessor = createGetterGeneric(propInfo);
+                    return accessor;
```

Comment:
```suggestion
                    return createGetterGeneric(propInfo);
```

Reply:
These all match the pattern that is used in this file. Granted, I touch a lot of the file, but `GetMakeNewInstanceFunc` still uses this pattern. I think the CoreFX guidelines say to keep existing pattern. But it's true the reasons for this pattern disappeared when debuggers started showing "The method returned X".